### ISSUE TITLE
Convert simple tables to Markdown tables

### ISSUE THREAD

**Issue by evbacher** [evbacher/gd2md-html#75]
Currently, Docs to Markdown converts all tables to HTML tables, which is also valid in Markdown files. But some simple tables could be converted to Markdown tables, and they would be more readable in source form.

We could probably start converting all tables to both Markdown and HTML, and if we run into something that is too complex (almost anything is too complex for Markdown tables), then we render the HTML table (and stop constructing the Markdown table).

This is complicated though -- will require some thought about what will trigger the change to HTML-only. If anyone wants to explore this, let me know.

**Comment by rgschmitz1** [comment]
I just started using your add-on, works great for the most part!  I actually just wrote a small bash script for generating the TOC before spotting this issue.  I'll take a look at your code and contribute if I can wrap my head around it.

**Comment by evbacher** [comment]
@rgschmitz1: Are you generating a Markdown TOC from the headings? That's a different issue from the Markdown tables, but also an interesting one. I have not thought about the Markdown/HTML table issue for a while, but it's certainly an interesting problem. (Though I have not pursued it much because the HTML tables can handle any kind of formatting, while Markdown tables are rather limited.)

**Comment by rgschmitz1** [comment]
Oops, I misunderstood this issue. Markdown tables do seem pretty limited in GitHub at least.

I am generating the TOC from headers, thankfully your tool left a clear marker ([TOC]). There's a bunch of scripts for this but I was hoping to make something a little less complicated for personal use.

**Comment by evbacher** [comment]
Yeah, when I originally wrote this add-on, we were generating a TOC server-side (via some JavaScript), triggered by the [TOC] directive. I don't have access to that server or code, but I suppose it would be possible to do something in the add-on itself based on the headings. Maybe I'll create a separate issue for that one.

**Comment by TheOtherBrian1** [comment]
It's been quite a while, is this issue still open?

**Comment by evbacher** [comment]
Sure -- I have not spent much thought on this recently, but if you have some ideas, this bug is still open. Check out https://github.com/evbacher/gd2md-html/blob/master/dev.md -- specifically on the testing and verification. It's best if  changes and enhancements do not break any existing functionality or create any obvious new bugs.

One other issue is if attempting to create Markdown tables significantly increases the computation time, that's not a good thing either, though it would probably only affect docs that are table-heavy. Worth testing though.

**Comment by ciriousjoker** [comment]
I'd also like to have this feature. When creating a very simple table, I'd like to convert it to the "|" table markdown syntax so ChatGPT understands it (supposedly!) better.

**Comment by linhaiqd** [comment]
There are no complicated tables in my document, just tables for taking notes. I really need this feature. Thank you.

**Comment by birdie60** [comment]
+1 for Markdown table support

**Comment by evbacher** [comment]
While Markdown tables seem simple, converting them from a Google Doc is not simple, so this is not an easy enhancement. We are working on some enhancements, and this is on the list, but I cannot promise that it will be a quick fix.

Thanks for your patience! (And remember that HTML is valid Markdown.)
